flo-runner/src/main/java/com/spotify/flo/context/ForkingEvalContext.java

```diff
@@ -0,0 +1,319 @@
+        } catch (InterruptedException e) {
+          process.destroyForcibly();
+          Thread.currentThread().interrupt();
+          throw new RuntimeException(e);
+        }
+        if (!exited) {
+          process.destroyForcibly();
+          throw new RuntimeException("Subprocess timed out");
+        }
+
+        if (process.exitValue() != 0) {
+          throw new RuntimeException("Subprocess failed: " + process.exitValue());
+        }
+
+        if (Files.exists(errorFile)) {
+          // Failed
+          final Throwable error;
+          try {
+            error = PersistingContext.deserialize(errorFile);
+          } catch (Exception e) {
+            throw new RuntimeException("Failed to deserialize error", e);
+          }
+          if (error instanceof RuntimeException) {
+            throw (RuntimeException) error;
+          } else {
+            throw new RuntimeException(error);
+          }
+        } else {
+          // Success
+          final T result;
+          try {
+            result = PersistingContext.deserialize(resultFile);
+          } catch (Exception e) {
+            throw new RuntimeException("Failed to deserialize result", e);
+          }
+          return result;
+        }
+      } finally {
+        executor.shutdownNow();
+        tryDeleteDir(tempdir);
+      }
+    };
+  }
+
+  private void tryDeleteDir(Path path) {
+    try {
+      deleteDir(path);
+    } catch (IOException e) {
+      LOG.warn("Failed to delete directory: {}", path, e);
+    }
+  }
+
+  private void deleteDir(Path path) throws IOException {
+    Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+        Files.delete(file);
+        return FileVisitResult.CONTINUE;
+      }
+
+      @Override
+      public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+        Files.delete(dir);
+        return FileVisitResult.CONTINUE;
+      }
+    });
+  }
+
+  private void copy(InputStream in, OutputStream out) {
+    byte[] buffer = new byte[1024];
+    try {
+      while (true) {
+        int r = in.read(buffer);
+        if (r < 0) {
+          break;
+        }
+        out.write(buffer, 0, r);
+        out.flush();
+      }
+    } catch (IOException e) {
+      log.error("Caught exception during byte stream copy", e);
+    }
+  }
+
+  @SuppressWarnings("finally")
+  private static class Trampoline {
+
+    private static final boolean DEBUG = Boolean.parseBoolean(System.getenv("FLO_DEBUG_FORKING"));
+
+    private static final String NAME = ManagementFactory.getRuntimeMXBean().getName();
+
+    private static class Watchdog
+        extends Thread {
+
+      Watchdog() {
+        setDaemon(false);
+      }
+
+      @Override
+      public void run() {
+        // Wait for parent to exit.
+        try {
+          while (true) {
+            int c = System.in.read();
```

Comment:
If parent process exits, here we will get EOF?

Reply:
Yeah